core/shared/src/main/scala/fs2/Chunk.scala

```diff
@@ -365,6 +365,12 @@ abstract class Chunk[+O] extends Serializable with ChunkPlatform[O] with ChunkRu
   def toByteVector[B >: O](implicit ev: B =:= Byte): ByteVector =
     this match {
       case c: Chunk.ByteVectorChunk => c.toByteVector
+      case c: Chunk.ByteBuffer      =>
+        // if c is a view, position/limit of underlying buffer may have changed
+        val bb = c.buf.asReadOnlyBuffer()
+        bb.position(c.offset)
```

Comment:
Do these need to take in to account the original position and limit as well? E.g. `bb.position(bb.position + c.offset)`

Reply:
I don't think so. `bb.position()` is not constant and `bb.position(i)` is not relative as far as I can make out.

Reply:
Okay got it, when we create a `Chunk.ByteBuffer`, we copy the position & limit in to offset & length.